src/metar/metar_csv2ioda.py

```diff
@@ -0,0 +1,322 @@
+
+import meteo_utils
+import ioda_conv_ncio as iconv
+from orddicts import DefaultOrderedDict
+
+class reformatMetar(object):
+
+    def __init__(self, filename, date):
+
+        self.filename = filename
+        self.date = date
+        self.meteo_utils = meteo_utils.meteo_utils()
+
+        # Read in CSV-formatted file of METAR data
+        self._rd_metars()
+
+        return
+
+    def _rd_metars(self):
+
+        # data is the dictionary of incoming observation (METAR) data
+        data = {}
+
+        data['ob_icao'] = []
+        data['ob_lat'] = []
+        data['ob_lon'] = []
+        data['ob_time'] = []
+        data['ob_datetime'] = []
+        data['ob_elev'] = []
+        data['ob_psfc'] = []
+        data['ob_temp'] = []
+        data['ob_spfh'] = []
+        data['ob_uwnd'] = []
+        data['ob_vwnd'] = []
+
+        '''
+        Read in the METARs data
+        Header contains: ICAO, Unix_time, Latitude, Longitude, Elev, MSLP, Altimeter, Temp, Dewp, Wdir, Wspd
+        '''
+
+        # open file in read mode
+        with open(self.filename, 'r') as fh:
+            # pass the file object to reader() to get the reader object
+            csv_dict_reader = DictReader(fh)
+            column_names = csv_dict_reader.fieldnames
+            print(column_names)
+            # Iterate over each row in the csv using reader object
+            for row in csv_dict_reader:
+                # row variable is a list that represents a row in csv
+
+                if row['ICAO'] is '':
+                    continue
+                else:
+                    icao = str(row['ICAO'])
+                try:
+                    utime = int(row['Unix_time'])
+                except:
+                    continue
+                try:
+                    lat = float(row['Latitude'])
+                except:
+                    continue
+                try:
+                    lon = float(row['Longitude'])
+                except:
+                    continue
+                try:
+                    elev = float(row['Elev'])
+                except:
+                    continue
+                try:
+                    temp = float(row['Temp']) + self.meteo_utils.C_2_K
+                except:
+                    temp = netCDF4.default_fillvals['f4']
+                try:
+                    dewp = float(row['Dewp']) + self.meteo_utils.C_2_K
+                except:
+                    dewp = netCDF4.default_fillvals['f4']
+                try:
+                    wdir = float(row['Wdir'])
+                except:
+                    wdir = netCDF4.default_fillvals['f4']
+                try:
+                    wspd = float(row['Wspd']) * self.meteo_utils.KTS_2_MS
+                except:
+                    wspd = netCDF4.default_fillvals['f4']
+
+                if ( (wdir is not netCDF4.default_fillvals['f4']) and (wspd is not netCDF4.default_fillvals['f4']) ):
+                    if (wdir==0 and wspd==0):
```

Comment:
Do we need this if, can't the trig function below handle this case?

Reply:
I believe the code is needed as-is, because it protects against the possibility (though really highly unlikely) that either wind speed *or* direction is missing while the other is not missing.  It shouldn't be possible, however, the apparent double-assignment of missing ensures there's no way to have one without the other - they both get forced to missing.

Reply:
I was referring to the if wdir==0 and wspd==0, not the is not missing

Reply:
Ah, now I got what you meant.  I coded it with efficiency in mind.  The first IF-test will rapidly get the result without the overhead of invoking the function - so this was done purposefully as a matter of speed.  I prefer it this way.

Reply:
I'm fine with leaving it in as is then, mainly as it's more readable/explicit, but if statements aren't free either, so the overhead will likely be minimal.

Reply:
Agreed that an IF-block is never 'free' either, however, the block for IF will always be required anyway whereas the function call can be optional in the specific case of (0,0) values.  So there is a minute efficiency improvement as coded, not just being more explicit and readable.